File: src/GroundItem.cpp
#include "headers/GroundItem.h"

GroundItem::GroundItem() {

}

GroundItem::GroundItem(int x, int y, sf::Texture* texture, sf::IntRect textureRect, int mods[3], bool stackable, int count, float weight) : Entity(x, y) {
    staticSprite = sf::Sprite(*texture, textureRect);

    width = 16;
    height = 16;

    healthMod = mods[0];
    defenseMod = mods[1];
    attackMod = mods[2];

    this->stackable = stackable;
    this->count = count;
    this->weight = weight;
}

void GroundItem::knockBack(float x, float y) {
    
}

void GroundItem::tick(std::vector<Entity*>* entities) {
    
}

void GroundItem::render(sf::RenderWindow* window, int xCameraOffset, int yCameraOffset) {
    staticSprite.setPosition(x - xCameraOffset, y - yCameraOffset);
    window->draw(staticSprite);
}

void GroundItem::renderToolTip(sf::RenderWindow* window, int xCameraOffset, int yCameraOffset) {
    tooltip.setPosition(x - 24 - xCameraOffset, y + 24 - yCameraOffset);
    window->draw(tooltip);
    tooltipText.setPosition(x - 20 - xCameraOffset, y + 20 - yCameraOffset);
    window->draw(tooltipText);
}

bool GroundItem::isStackable() {
    return stackable;
}

int GroundItem::getCount() {
    return count;
}

int* GroundItem::getStatMods() {
     int statMods[3] = {healthMod, defenseMod, attackMod};
     return statMods;
}

float GroundItem::getWeight() {
    return weight;
}

File: src/Entities/GroundItem.cs
using SFML.Graphics;
using SFML.System;

namespace TAC {
    class GroundItem : Entity {

        public Item ItemReference {get; set;}
        public bool PickedUp {get; set;} = false;

        private RectangleShape tooltipBackground;
        private Text tooltipText;

        public GroundItem(Item itemRef, float x, float y) : base() {
            ItemReference = itemRef;

            MaxHealth = 1;
            Health = 1;
            DisplayHealth = 1;

            X = x;
            Y = y;

            EntitySprite = new Sprite(Assets.items);
            EntitySprite.TextureRect = itemRef.Icon.TextureRect;

            tooltipBackground = new RectangleShape(new Vector2f(96.0f, 48.0f));
            tooltipBackground.FillColor = new Color(36, 58, 71);
            tooltipBackground.OutlineColor = Color.Black;
            tooltipBackground.OutlineThickness = 1.0f;

            tooltipText = new Text(ItemReference.Name + "\nValue: " + ItemReference.Value + "\nWeight:" + ItemReference.Weight, Assets.defaultFont);
            tooltipText.FillColor = new Color(200, 200, 200);
            tooltipText.OutlineColor = Color.Black;
            tooltipText.OutlineThickness = 1.0f;
            tooltipText.CharacterSize = 15;

            tick += () => {

            };

            render += (RenderWindow window) => {
                
            };
        }

        public void renderTooltip(RenderWindow window) {
            if (!Hovered) return;

            //render tooltip if hovered
            tooltipBackground.Position = new Vector2f((int)(EntitySprite.Position.X + EntitySprite.TextureRect.Width), (int)(EntitySprite.Position.Y + EntitySprite.TextureRect.Height));
            window.Draw(tooltipBackground);
            tooltipText.Position = new Vector2f(tooltipBackground.Position.X + 2, tooltipBackground.Position.Y);
            window.Draw(tooltipText);
        }
    }
}